sound/soc/sof/intel/hda.c

```diff
@@ -185,6 +185,10 @@ static int hda_sdw_probe(struct snd_sof_dev *sdev)
 
 	memset(&res, 0, sizeof(res));
 
+	/* sdw trigger callback is only needed for IPC4 */
+	if (sdev->pdata->ipc_type == SOF_INTEL_IPC4)
+		sdw_callback.trigger = ipc4_be_dai_trigger;
```

Comment:
I see, we are re-using the existing `ipc4_be_dai_trigger()`, that's why the different sets of parameters for the trigger callback.